src/Block.cpp

```diff
@@ -188,14 +190,40 @@ Block::make_random(CGContext &cg_context, bool looping)
 
 	curr_func->stack.pop_back();
 	if (Error::get_error() != SUCCESS) {
-		//curr_func->stack.pop_back();
+		curr_func->stack.pop_back();
 		delete b;
 		return NULL;
 	}
 
 	// ISSUE: in the exhaustive mode, do we need a return statement here
 	// if the last statement is not?
 	Error::set_error(SUCCESS);
+
+
+	if(CGOptions::computed_goto()){
```

Comment:
In Csmith, the `make_random` functions defined in various classes generate an AST node of the class. `StatementGoto::make_random` generates a goto-statement, and if we add `alt_label` as a field to `StatementGoto`, we expect `alt_label` not to be altered AFTER  `StatementGoto::make_random`  is called.  
  
`stm_labels` maps all destinations of goto-statement (as a Statement*) to their labels. It's there to avoid we assign two labels to the same destination.  
  
`addr_label` stores labels for a given block. The static map in StatementGoto.h would map all destinations of goto-statement (as Statement*) to their alternative labels in the whole program. This static map is not part of the `StatementGoto` object (the AST node), thus we are fine to modify it after the block is finalized.